llvm/lib/Transforms/InstCombine/InstCombineCalls.cpp

```diff
@@ -1633,6 +1667,21 @@ Instruction *InstCombinerImpl::visitCallInst(CallInst &CI) {
       Value *Cmp = Builder.CreateICmpNE(I0, Zero);
       return CastInst::Create(Instruction::ZExt, Cmp, II->getType());
     }
+    // umin(cttz(x), const) --> cttz(x | (1 << const))
+    Value *X;
+    Constant *Y;
+    Value *Z;
+    if (match(I0, m_OneUse(m_Cttz(m_Value(X), m_Value(Z)))) &&
+        match(I1, m_Constant(Y))) {
```

Comment:
```suggestion
        match(I1, m_ImmConstant(Y))) {
```
Otherwise this will crash and burn with constant expressions.

Reply:
I've replaced the matcher with `m_CheckedInt`.